pkg/controller/common/manifestprocessing.go

```diff
@@ -40,35 +40,36 @@ func NewManifestProcessor(controllerResources ControllerResources, appInstance,
 	}
 }
 
-func (p *ManifestProcessor) ProcessManifests(manifests []manifest.Manifest, component string) error {
+func (p *ManifestProcessor) ProcessManifests(ctx context.Context, manifests []manifest.Manifest, component string) error {
 	allErrors := []error{}
 
-	origLogger := p.Log
-	defer func() { p.Log = origLogger }()
+	origCtx := ctx
+	origLogger := LogFromContext(ctx)
```

Comment:
I don't think this is necessary, if we redeclare ctx in the loop below.

Reply:
I had to extract a method to get rid of this. IMHO, it makes sense to extract a function every time a section of code has its own logger and context. Interesting how logging can drive the structuring of code.